File: jwt/src/main/java/com/member/jwt/repository/UserRepository.java
package com.member.jwt.repository;

import com.member.jwt.entity.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserRepository extends JpaRepository<UserEntity, Integer> {

    Boolean existsByUsername(String username);

    // username을 받아 DB 테이블에서 회원 조회하는 메서드 작성
    UserEntity findByUsername(String username);
}

File: jwtKotlin/src/main/kotlin/com/member/jwtkotlin/repository/UserRepository.kt
package com.member.jwtkotlin.repository

import com.member.jwtkotlin.entity.UserEntity
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository : JpaRepository<UserEntity, Int> {
    fun existsByUsername(username: String): Boolean

    // username을 받아 DB 테이블에서 회원 조회하는 메서드
    fun findByUsername(username: String): UserEntity?
}
